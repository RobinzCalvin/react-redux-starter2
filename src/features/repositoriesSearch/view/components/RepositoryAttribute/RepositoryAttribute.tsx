import React from 'react';
import block from 'bem-cn';

import './RepositoryAttribute.scss';

interface IProps {
  title: string;
  value: string | number;
  type?: 'owner';
  onValueClick?(): void;
}

const b = block('repository-attribute');

function RepositoryAttribute(props: IProps) {
  const { title, value, onValueClick, type } = props;
  const handleValueKeyPress = (e: React.KeyboardEvent<HTMLSpanElement>) => {
    if (e.key === 'Enter' && onValueClick) {
      onValueClick();
    }
  };

  return (
    <div className={b()}>
      <span className={b('title')}>
        {title}
      </span>
      <span
        tabIndex={0}
        role="button"
        className={b('value', { type })}
        onClick={onValueClick}
        onKeyPress={handleValueKeyPress}
      >
        {value}
      </span>
    </div>
  );
}

export { RepositoryAttribute, IProps as IRepositoryAttributeProps };
