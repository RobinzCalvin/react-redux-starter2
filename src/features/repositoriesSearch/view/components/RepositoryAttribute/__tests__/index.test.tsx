import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import RepositoryAttribute, { IRepositoryAttributeProps } from '../RepositoryAttribute';

const props: IRepositoryAttributeProps = {
  title: 'title',
  value: 'value',
  onValueClick: jest.fn(),
};

describe('(features/repositoriesSearch) RepositoryAttribute component', () => {
  let component: ShallowWrapper<IRepositoryAttributeProps>;
  beforeEach(() => component = shallow(<RepositoryAttribute {...props} />));

  it('should call onValueClick on value click', () => {
    component.find('.repository-attribute__value').simulate('click');
    expect(props.onValueClick).toHaveBeenCalledTimes(1);
  });
});
