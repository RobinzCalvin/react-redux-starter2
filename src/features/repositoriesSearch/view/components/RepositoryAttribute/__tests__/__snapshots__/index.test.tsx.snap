// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryAttribute component should match snapshot 1`] = `
<div
  className="repository-attribute"
>
  <span
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
  >
    title
    :
  </span>
  <span
    className={
      Object {
        "has": [Function],
        "is": [Function],
        "mix": [Function],
        "split": [Function],
        "state": [Function],
        "toString": [Function],
      }
    }
    onClick={[MockFunction]}
  >
    value
  </span>
</div>
`;
